thorlcr/graph/thgraph.cpp

```diff
@@ -2691,6 +2699,11 @@ void CJobChannel::addDependencies(IPropertyTree *xgmml, bool failIfMissing)
         }
         CGraphElementBase *targetActivity = (CGraphElementBase *)target->queryElement(edge.getPropInt("att[@name=\"_targetActivity\"]/@value"));
         CGraphElementBase *sourceActivity = (CGraphElementBase *)source->queryElement(edge.getPropInt("att[@name=\"_sourceActivity\"]/@value"));
+        if (TAKlocalresultwrite == sourceActivity->getKind() && (TAKlocalresultread != targetActivity->getKind()))
+        {
+            if (target->isGlobal() && source->isLoopSubGraph())
```

Comment:
I would have expected source and target to be switched in the code below.  Can you explain the logic?